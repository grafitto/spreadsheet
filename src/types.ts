import { FunctionCallExpression, IdentifierExpression, ParametersExpression } from './compiler/expressions';

export interface Cell {
  formula: string;
  value: number | string;
}

export interface AppState {
  grid: Grid;
}

export type Grid = Array<Array<Cell>>;

export type ParameterExpression = (FunctionCallExpression | ParametersExpression | IdentifierExpression)[]
export interface Payload {
  cell: {
    row: number;
    column: number;
  };
  value: {
    formula: string;
    value: string | number;
  };
}