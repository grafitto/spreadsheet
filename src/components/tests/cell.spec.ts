import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CellView from '../cell.vue';
import { Grid } from '@/types';
import { initialiseGrid } from '@/helpers/grid';

const localVue = createLocalVue()
localVue.use(Vuex);

function createStore (overrides?: any) {
  const defaults = {
    state: { grid: initialiseGrid() }
  }
  const store = new Vuex.Store({...defaults, ...overrides});
  return store;
}

function createWrapper (overrides?: any) {
  const defaults = {
    store: createStore()
  }
  return shallowMount(CellView, {
    ...{ ...defaults, ...overrides },
    localVue,
    propsData: {
      column: 12,
      row: 12
    }
  });
}

describe('Cell.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = createWrapper();
  })

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should make sure input exists', () => {
    expect(wrapper.findAll('input').length).toEqual(1);
  })

  afterEach(() => wrapper.destroy())
})