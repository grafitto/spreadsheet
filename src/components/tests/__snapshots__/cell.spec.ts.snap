// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell.vue should render correctly 1`] = `<input>`;
